pkg/controller/infinispan/resources/operatorconfig/grafana.go

```diff
@@ -0,0 +1,95 @@
+package operatorconfig
+
+import (
+	"github.com/infinispan/infinispan-operator/pkg/controller/constants"
+	consts "github.com/infinispan/infinispan-operator/pkg/controller/constants"
+	kube "github.com/infinispan/infinispan-operator/pkg/kubernetes"
+	grafanav1alpha1 "github.com/integr8ly/grafana-operator/v3/pkg/apis/integreatly/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	grafanaDashboardNamespace = "grafana.dashboard.namespace"
+)
+
+// reconcileGrafana reconciles grafana object status with the operator configuration settings
+func (r *ReconcileOperatorConfig) reconcileGrafana(config map[string]string, currentConfig map[string]string, operatorNs string) (*reconcile.Result, error) {
+	grafanaNs := config[grafanaDashboardNamespace]
+	currGrafanaNs := currentConfig[grafanaDashboardNamespace]
+	// Delete current grafana dashboard if namespace is changes
+	if grafanaNs != currGrafanaNs && currGrafanaNs != "" {
+		currGrafana := &grafanav1alpha1.GrafanaDashboard{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      constants.GetEnvWithDefault(ApplicationNameVariable, DefaultApplicationName),
+				Namespace: currGrafanaNs,
+			},
+		}
+		if err := r.Kube.Client.Delete(ctx, currGrafana); err != nil && !errors.IsNotFound(err) {
+			return &reconcile.Result{}, err
+		}
+	}
+	currentConfig[grafanaDashboardNamespace] = ""
+	if grafanaNs == "" {
+		return &reconcile.Result{}, nil
+	}
+	// detect the GrafanaDashboard resource type resourceExists on the cluster
+	resourceExists, err := r.Kube.IsGroupVersionSupported(grafanav1alpha1.SchemeGroupVersion.String(), grafanav1alpha1.GrafanaDashboardKind)
+	if err != nil {
+		r.Log.Error(err, "Error checking Grafana support")
+		return &reconcile.Result{Requeue: true}, nil
```

Comment:
We should add a delay with this operation, as adding Grafana operator with CRD's is a customer facing action